### ISSUE TITLE
Wrong error message when attach to a non-existing container

### ISSUE THREAD

**Issue by jybuestc** [docker/docker#21983]
---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
Use the commands below to provide key information from your environment:
You do NOT have to include this information if this is a FEATURE REQUEST
-->

**Output of `docker version`:**
Client:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   20f81dd
 Built:        Thu Mar 10 15:39:25 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   20f81dd
 Built:        Thu Mar 10 15:39:25 2016
 OS/Arch:      linux/amd64

**Output of `docker info`:**

```
Containers: 9
 Running: 4
 Paused: 0
 Stopped: 5
Images: 11
Server Version: 1.10.3
Storage Driver: devicemapper
 Pool Name: docker-253:0-1076567761-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 2.041 GB
 Data Space Total: 107.4 GB
 Data Space Available: 105.3 GB
 Metadata Space Used: 4.305 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.143 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Either use `--storage-opt dm.thinpooldev` or use `--storage-opt dm.no_warn_on_loop_devices=true` to suppress this warning.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.107-RHEL7 (2015-12-01)
Execution Driver: native-0.2
Logging Driver: json-file
Plugins:
 Volume: local
 Network: host bridge null
Kernel Version: 3.10.0-327.13.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 16
Total Memory: 94.25 GiB
Name: nchyp194104.hyperv.drm.lab.emc.com
ID: I2JE:IPBE:KXRN:AWGB:IS7X:XNZ6:D3TJ:SZJ2:724P:JBIZ:L7DM:3SAP
Username: yangbo1234
Registry: https://index.docker.io/v1/
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled

```

**Steps to reproduce the issue:**
1. Try to attach a non-existing container , it failed with improper error message

**Describe the results you received:**

[root@nchyp194104 ~]# docker attach deb085f13866
You cannot attach to a stopped container, start it first

**Describe the results you expected:**

You cannot attach a non-existing container.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Comment by AkihiroSuda** [comment]
I can reproducible with docker 1.10.3, but could not reproduce with 1.11-rc4.
I guess it is already fixed in some commit.

**Comment by AkihiroSuda** [comment]
Could not reproduce with B2D 1.10.3 (aufs)

**Comment by jybuestc** [comment]
Hi Suda, what error message you received?

**Comment by AkihiroSuda** [comment]
@jybuestc 
```
docker@dm1:~$ docker ps -a
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
```

```
docker@dm1:~$ docker attach deb085f13866
Error: No such container: deb085f13866
```

```
docker@dm1:~$ docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: 1.10.3
Storage Driver: devicemapper
 Pool Name: docker-8:1-261689-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: ext4
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 305.7 MB
 Data Space Total: 107.4 GB
 Data Space Available: 19.15 GB
 Metadata Space Used: 729.1 kB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.147 GB
 Udev Sync Supported: false
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /mnt/sda1/var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Either use `--storage-opt dm.thinpooldev` or use `--storage-opt dm.no_warn_on_loop_devices=true` to suppress this warning.
 Metadata loop file: /mnt/sda1/var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.82 (2013-10-04)
Execution Driver: native-0.2
Logging Driver: json-file
Plugins: 
 Volume: local
 Network: bridge null host
Kernel Version: 4.1.19-boot2docker
Operating System: Boot2Docker 1.10.3 (TCL 6.4.1); master : 625117e - Thu Mar 10 22:09:02 UTC 2016
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 996.1 MiB
Name: dm1
ID: K3AN:OEGS:3ENX:7SKV:JR3P:WP66:ROZ5:2TBF:UVBU:RQ6Z:QIWT:4PLU
Debug mode (server): true
 File Descriptors: 11
 Goroutines: 21
 System Time: 2016-04-13T09:10:47.248518932Z
 EventsListeners: 0
 Init SHA1: 
 Init Path: /usr/local/bin/docker
 Docker Root Dir: /mnt/sda1/var/lib/docker
Labels:
 provider=virtualbox
```

```
docker@dm1:~$ docker version
Client:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   20f81dd
 Built:        Thu Mar 10 21:49:11 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   20f81dd
 Built:        Thu Mar 10 21:49:11 2016
 OS/Arch:      linux/amd64
```

**Comment by thaJeztah** [comment]
I'm not able to reproduce this as well; I suspect the container still exists but is stopped; try running `docker ps -a` to verify if the container really doesn't exist.

I'm closing this issue, because I suspect this is not a bug, but please comment with more information if you think this is a bug
